bundles.rst

```diff
@@ -48,33 +48,52 @@ The new bundle is called AcmeTestBundle, where the ``Acme`` portion is an exampl
 name that should be replaced by some "vendor" name that represents you or your
 organization (e.g. ABCTestBundle for some company named ``ABC``).
 
-Start by creating a ``src/Acme/TestBundle/`` directory and adding a new file
+Start by creating a ``Acme/TestBundle/src/`` directory and adding a new file
 called ``AcmeTestBundle.php``::
 
-    // src/Acme/TestBundle/AcmeTestBundle.php
-    namespace App\Acme\TestBundle;
+    // Acme/TestBundle/src/AcmeTestBundle.php
+    namespace Acme\TestBundle;
 
-    use Symfony\Component\HttpKernel\Bundle\Bundle;
+    use Symfony\Component\HttpKernel\Bundle\AbstractBundle;
 
-    class AcmeTestBundle extends Bundle
+    class AcmeTestBundle extends AbstractBundle
     {
     }
 
+.. versionadded:: 6.1
+
+    The ``AbstractBundle`` was introduced in Symfony 6.1. If your bundle must be compatible
+    with previous Symfony versions you have to extend from the :class:`Symfony\\Component\\HttpKernel\\Bundle\\Bundle`
+    instead.
+
 .. tip::
 
     The name AcmeTestBundle follows the standard
     :ref:`Bundle naming conventions <bundles-naming-conventions>`. You could
     also choose to shorten the name of the bundle to simply TestBundle by naming
     this class TestBundle (and naming the file ``TestBundle.php``).
 
-This empty class is the only piece you need to create the new bundle. Though
+It's recommended to place your bundle class in the ``src/`` directory and keep out all the
+configuration files, templates, translations, etc. By default, Symfony determines the bundle path from the
+directory where the bundle class is placed, so you have to define the :method:`Symfony\\Component\\HttpKernel\\Bundle\\Bundle::getPath`
+method to tell Symfony what is the root directory of your bundle path::
+
+    class AcmeTestBundle extends AbstractBundle
+    {
+        public function getPath(): string
+        {
+            return \dirname(__DIR__);
+        }
```

Comment:
I would like to change the default convention to not be forced to adjust the bundle path. The recommended bundle directory structure is already supported since Symfony 4.4, so we could take advantage of this new class to change the default path without BC breaks.

Reply:
Yes please! 🙌 
Was always a pet peeve

Reply:
Done in https://github.com/symfony/symfony/pull/46385 (waiting for review/merge)

Reply:
Note/Code removed. Now when your bundle extends from `AbstractBundle` Symfony assumes you're using the modern directory structure.